lib/tasks/data_maintenance.rake

```diff
@@ -62,4 +62,21 @@ namespace :data_maintenance do
     GenericEvent.where(type: incident_event_types).update_all(classification: 'incident')
     GenericEvent.where(type: notification_event_types).update_all(classification: 'notification')
   end
+
+  desc 'remove moves and allocations on staging'
+  task remove_data_staging: :environment do
+    # NB: this task should never run on production or pre-production
+    is_not_production = Rails.env.development? || ENV.fetch('HOSTNAME', 'UNKNOWN') =~ /(\-(dev|staging|uat)\-)/i
+
+    if is_not_production
+      moves = Move.all
+      puts "Number of moves to be deleted: #{moves.count}"
+      allocations = Allocation.all
+      puts "Number of allocations to be deleted: #{allocations.count}"
+      moves.destroy_all
```

Comment:
I think this won't work as we would get `ActiveRecord::InvalidForeignKey` if anything is associated to moves, for instance assessments or journeys etc.

Reply:
Ah, would it be safer to do a delete_all then? I took the destroy all from a task further up this file which uses it moves, but perhaps have missed some context.

Reply:
I tried running both `destroy_all` and `delete_all` locally to see which one ignores callbacks, as `delete_all` might be the one without callbacks, but they both threw the error because it had a youth risk assessment associated to it 🤔